libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -151,6 +151,40 @@ void ArduinoOTAClass::begin(bool useMDNS) {
 #endif
 }
 
+/**
+ * Generates nonce using PRNG / TRNG if available
+ * @return  Random MD5 string (nonce)
+ */
+String ArduinoOTAClass::generateNonce(void) {
```

Comment:
This method has no dependency on the ArudinoOTA class members or methods. Therefore, it doesn't make sense to make it a class method. That just pollutes the class interface.
Please make it a static function private to this file, or better yet, move it to an anonymous namespace at the beginning of this .cpp.